Index: \vue-web-component-wrapper\package\dist\vue-web-component-wrapper.es.js
===================================================================
--- \vue-web-component-wrapper\package\dist\vue-web-component-wrapper.es.js
+++ \vue-web-component-wrapper\package\dist\vue-web-component-wrapper.es.js
@@ -41,9 +41,9 @@
     r.mixin({
       mounted() {
         var n;
         const t = (i) => {
-          i != null && i.length && (this.__style = document.createElement("style"), this.__style.innerText = i.join().replace(/\n/g, ""), b(this.$el).prepend(this.__style));
+          i != null && i.length && (this.__style = document.createElement("style"), this.__style.innerText = i.join().replace(/\n/g, ""), b(this.$el).prepend(this.__style), b(this.$el).parentNode?.host?.append(this.__style));
         };
         if (t((n = this.$) == null ? void 0 : n.type.styles), this.$options.components)
           for (const i of Object.values(this.$options.components))
             t(i.styles);
Index: \vue-web-component-wrapper\package\src\web-component-util.js
===================================================================
--- \vue-web-component-wrapper\package\src\web-component-util.js
+++ \vue-web-component-wrapper\package\src\web-component-util.js
@@ -56,9 +56,11 @@
           const insertStyles = (styles) => {
             if (styles?.length) {
               this.__style = document.createElement('style')
               this.__style.innerText = styles.join().replace(/\n/g, '')
-              nearestElement(this.$el).prepend(this.__style)
+              const el = nearestElement(this.$el)
+              el.prepend(this.__style)
+              el.parentNode?.host?.append(this.__style);
             }
           }
 
           insertStyles(this.$?.type.styles)
